test/metabase/task/send_pulses_test.clj

```diff
@@ -88,13 +89,34 @@
                                                         :channel_type :slack
                                                         :details      {}}
                                                        daily-at-1am)]
-          (#'task.send-pulses/clear-pcs-and-send-pulse! pulse #{pc pc-disabled pc-no-recipient})
+          (#'task.send-pulses/send-pulse!* daily-at-1am pulse #{pc pc-disabled pc-no-recipient})
           (testing "only send to enabled channels that has recipients"
             (is (= #{pc} @sent-channel-ids)))
 
           (testing "channels that has no recipients are deleted"
             (is (false? (t2/exists? :model/PulseChannel pc-no-recipient)))))))))
 
+(deftest send-pulse!*-update-trigger-priority-test
+  (testing "send-pulse!* should update the priority of the trigger based on the duration of the pulse"
+    (pulse-channel-test/with-send-pulse-setup!
+     (with-redefs [metabase.pulse/send-pulse! (fn [& _args]
+                                                ;; priority is duration round to seconds
+                                                (Thread/sleep 1000))]
+       (mt/with-temp
+         [:model/Pulse        {pulse :id} {}
+          :model/PulseChannel {pc :id}    (merge
+                                           {:pulse_id     pulse
+                                            :channel_type :slack
+                                            :details      {:channel "#random"}}
+                                           daily-at-1am)]
+         (testing "priority is 0 to start with"
+           (is (= 6 (-> (pulse-channel-test/send-pulse-triggers pulse) first :priority))))
+         (#'task.send-pulses/send-pulse!* daily-at-1am pulse #{pc})
+         (testing "send pulse should update its priority"
+           ;; 5 is the default priority of a trigger, we need it to be higher than that because
+           ;; pulse is time sensitive compared to other tasks like sync
+           (is (> (-> (pulse-channel-test/send-pulse-triggers pulse) first :priority) 5))))))))
```

Comment:
It would be nice if we tested that the priority is (non-strictly) monotonically increasing based on duration.

Reply:
yea we can redef the `duration-ms->priority` now it's a function.